vulnerabilities/helpers.py

```diff
@@ -267,3 +267,81 @@ def _get_gh_response(gh_token, graphql_query):
     endpoint = "https://api.github.com/graphql"
     headers = {"Authorization": f"bearer {gh_token}"}
     return requests.post(endpoint, headers=headers, json=graphql_query).json()
+
+
+def get_reference_id(url: str):
+    """
+    Return the reference id from a URL
+    For example:
+    >>> get_reference_id("https://github.com/advisories/GHSA-c9hw-wf7x-jp9j")
+    'GHSA-c9hw-wf7x-jp9j'
+    """
+    url_parts = url.split("/")
+    last_url_part = url_parts[-1]
+    return last_url_part
+
+
+def extract_references(urls: List[str]):
+
+    # To avoid circular import
+    from vulnerabilities.importer import Reference
+
+    """
+    Yield `reference` by iterating over `reference_data`
+    >>> list(extract_references(['url': "https://github.com/advisories/GHSA-c9hw-wf7x-jp9j"]))
+    [Reference(url="https://github.com/advisories/GHSA-c9hw-wf7x-jp9j"), reference_id = "GHSA-c9hw-wf7x-jp9j" ]
+    >>> list(extract_references(['url': "https://github.com/advisories/c9hw-wf7x-jp9j"]))
+    [Reference(url="https://github.com/advisories/c9hw-wf7x-jp9j")]
+    """
+    for url in urls:
+        if not isinstance(url, str):
+            logger.error(f"extract_references: url is not of type `str`: {url}")
+            continue
+        reference_id = get_reference_id(url)
+        if "GHSA-" in reference_id.upper():
+            reference = Reference(url=url, reference_id=get_reference_id(url))
+        elif is_cve(reference_id):
+            reference = Reference(url=url, reference_id=reference_id.upper())
+        else:
+            reference = Reference(url=url)
+        yield reference
+
+
+def resolve_version_range(
+    affected_version_range: VersionRange,
+    package_versions: List[str],
+    ignorable_versions: List[str],
+) -> Tuple[List[str], List[str]]:
+    """
+    Given an affected version range and a list of `package_versions`, resolve
+    which versions are in this range and return a tuple of two lists of
+    `affected_versions` and `unaffected_versions`.
+    """
+    if not affected_version_range:
+        logger.error(f"affected version range is {affected_version_range!r}")
+        return [], []
+    affected_versions = []
+    unaffected_versions = []
+    for package_version in package_versions or []:
+        if package_version in ignorable_versions:
+            continue
+        # Remove whitespace
+        package_version = package_version.replace(" ", "")
+        # Remove leading 'v'
+        package_version = package_version.lstrip("vV")
```

Comment:
Should not these two removal be in some `version_class.from_string()` method of sorts?

Reply:
This is a shared helper function, so in case if a version_class does not strip 'v', this will help us there